src/tools/spin/spinquic.cpp

```diff
@@ -640,6 +641,18 @@ CXPLAT_THREAD_CALLBACK(ServerSpin, Context)
         }
     }
 
+    if (Settings.AllocFailDenominator > 0) {
```

Comment:
This is a good start, but as it skips everything above (same for other thread) it's not giving us code coverage on those alloc failures. Please put a `TODO` comment to move this to right after we call `MsQuicOpen` instead, and then update all these things to be failure proof.

Reply:
Yeah, that's exactly why I put them in the loop instead of after `MsQuicOpen`.  How should we handle these failures during setup? Retry? Exit?

Reply:
Just exit if it fails I think. Let's not make things too complicated. We have a retry for this at the higher layer already I think.